Modules/mathmodule.c

```diff
@@ -2016,6 +2016,78 @@ math_factorial(PyObject *module, PyObject *arg)
 }
 
 
+/*[clinic input]
+math.lcm
+
+	x as a: object
+	y as b: object
+    /
+least common multiple of x and y
+[clinic start generated code]*/
+
+static PyObject*
+math_lcm(PyObject*module,PyObject*a,PyObject*b)
+{
+    PyObject * g,*m,*f,*ab;
+    if (PyFloat_Check(a)||PyFloat_Check(b)) {
+    if (PyErr_WarnEx(PyExc_DeprecationWarning,
```

Comment:
It souds like a bad idea to add a new function directly with a deprecated feature. Why not simply directly rejecting floats?